lib/couchbase/cluster_registry.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require "singleton"
+
+require "couchbase/errors"
+
+module Couchbase
+  class ClusterRegistry
+    include Singleton
```

Comment:
Not sure if the use of `Singleton` is needed here. Initially I was thinking about having these methods here as 'static' and `@handlers` as a class instance variable, but having mutable class instance variables violates rubocop's rules